tests/tests_fabric/loggers/test_csv.py

```diff
@@ -30,6 +30,16 @@ def test_file_logger_automatic_versioning(tmpdir):
     assert logger.version == 2
 
 
+def test_file_logger_automatic_versioning_relative_root_dir(tmpdir):
+    """Verify that automatic versioning works, when root_dir is given a relative path."""
+    root_dir = tmpdir.mkdir("exp")
+    root_dir.mkdir("version_0")
+    root_dir.mkdir("version_1")
+    os.chdir(tmpdir)
+    logger = CSVLogger(root_dir="exp", name="exp")
```

Comment:
```suggestion
    logger = CSVLogger(root_dir="exp", name="name")
```

Reply:
@awaelchli  Suppose the logging path looks like `some_dir/exp/version_0`, this `root_dir` should be `some_dir/exp` according to its internal usage (more info at https://github.com/Lightning-AI/lightning/issues/17155). Though the name parameter is not used internally by the CSVLogger to calculate the next version, from my understanding it should refer to the `exp` directory. May I ask why we'd want to use `name="name"` instead?

Reply:
@awaelchli My point is that the `root_dir` should be equal to the `name` parameter because they refer to the same directory. If they have to be different, you can create a nested directory for testing, like `root_dir=dir1/dir2` and `name=dir2`.

Reply:
root_dir and name cannot refer to the same directory, that's not allowed. We need to guarantee the following folder structure: `root_dir/name/version`. 

So if you do `CSVLogger(root_dir="exp", name="exp")` we expect this `./exp/exp/version` to be the path where logs get saved.

Reply:
In your other example `CSVLogger(root_dir="dir1/dir2", name="dir2")` would result in a directory structure of `./dir1/dir2/dir2` which is expected.

Reply:
@awaelchli Yes, but unfortunately, that is the current behavior for pytorch.loggers.CSVLogger, but not fabric CSVLogger, and that issue is what I described in https://github.com/Lightning-AI/lightning/issues/17155.

Reply:
@awaelchli This issue has another fix, which will produce the correct behavior you described, and was my first attempt. The fix will add `save_dir=os.path.join(self.root_dir, self.name)` right after the first line of `_get_next_version`. But this will make all other tests about versions in `test_csv.py` fail, since other tests are passing `root_dir` the same as `name`

Reply:
@awaelchli Is it possible to get this bug fixed in whatever way you prefer? The bug itself is simple to spot and fix and is not entangled with any refactoring or design issues.